### PR TITLE
Add more enable switch for predicate plugin.

### LINKED ISSUE #2748: Add more enable switch in `predicate` plugin.
#### What would you like to be added:

In volcano, all `filter` functions of origin Kubernetes was integrated in `predicate` plugin. So we could not disable some of the `filter` functions like `NodePort` meanwhile enable the other functions like `NodeAffinity`. So we need more enable arguments in `predicate` plugin to switch these `filters`.

The enable arguments could be like:

```yaml
actions: "reclaim, allocate, backfill, preempt"
tiers:
- plugins:
  - name: priority
  - name: gang
  - name: conformance
- plugins:
  - name: drf
  - name: predicates
    arguments:
      predicate.NodeAffinityEnable: true
      predicate.NodePortsEnable: true
      predicate.TaintTolerationEnable: true
      predicate.podAffinityEnable: true
      predicate.NodeVolumeLimitsEnable: true
      predicate.VolumeZoneEnable: true
      predicate.PodTopologySpreadEnable: true
      predicate.GPUSharingEnable: true
      predicate.GPUNumberEnable: true
      predicate.CacheEnable: true
      predicate.ProportionalEnable: true
      predicate.resources: nvidia.com/gpu
      predicate.resources.nvidia.com/gpu.cpu: 4
      predicate.resources.nvidia.com/gpu.memory: 8
  - name: proportion
  - name: nodeorder
```

### PR DESCRIPTION
**Pull request by jiangkaihua**
Fix #2748

### CODE REVIEW

File: pkg/scheduler/plugins/predicates/predicates.go [hunk 1]
```diff
@@ -279,7 +331,9 @@ func (pp *predicatesPlugin) OnSessionOpen(ssn *framework.Session) {
 
 	ssn.AddPrePredicateFn(pp.Name(), func(task *api.TaskInfo) error {
 		// Check NodePorts
-		nodePortFilter.PreFilter(context.TODO(), state, task.Pod)
+		if predicate.nodePortEnable {
+			nodePortFilter.PreFilter(context.TODO(), state, task.Pod)
```
Comment (wangyang0616) [inline pkg/scheduler/plugins/predicates/predicates.go:335]: After adding `nodePortEnable` to judge, is it better to transfer the definition of `nodePortFilter` from the front to here? 

There are also other variable definitions, such as: `nodeaffinity`, `taintToleration`, `podAffinityFilter`, etc.

**Comment by william-wang** [comment]
@jiangkaihua please have a look at the UT failure.

**Review by william-wang — APPROVED** [review]
/lgtm
/approve

**Pull request merged**